src/main/java/org/apache/hadoop/mapred/ResourcePolicy.java

```diff
@@ -437,50 +440,58 @@ public void resourceOffers(SchedulerDriver schedulerDriver,
         overrides.set("mapred.task.tracker.report.address",
             reportAddress.getHostName() + ':' + reportAddress.getPort());
 
-        overrides.setLong("mapred.tasktracker.map.tasks.maximum",
-            mapSlots);
-
-        overrides.setLong("mapred.tasktracker.reduce.tasks.maximum",
-            reduceSlots);
-
+        overrides.setLong("mapred.tasktracker.map.tasks.maximum", mapSlots);
+        overrides.setLong("mapred.tasktracker.reduce.tasks.maximum", reduceSlots);
 
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        try {
-          overrides.write(new DataOutputStream(baos));
-          baos.flush();
-        } catch (IOException e) {
-          LOG.warn("Failed to serialize configuration.", e);
-          System.exit(1);
-        }
-
-        byte[] bytes = baos.toByteArray();
-
-        TaskInfo info = TaskInfo
+        // Build up the executor info
+        ExecutorInfo executor = ExecutorInfo
             .newBuilder()
-            .setName(taskId.getValue())
-            .setTaskId(taskId)
-            .setSlaveId(offer.getSlaveId())
+            .setExecutorId(ExecutorID.newBuilder().setValue(
+                "executor_" + taskId.getValue()))
+            .setName("Hadoop TaskTracker")
+            .setSource(taskId.getValue())
             .addResources(
                 Resource
                     .newBuilder()
                     .setName("cpus")
                     .setType(Value.Type.SCALAR)
                     .setRole(cpuRole)
-                    .setScalar(Value.Scalar.newBuilder().setValue(taskCpus - containerCpus)))
+                    .setScalar(Value.Scalar.newBuilder().setValue(containerCpus)))
             .addResources(
                 Resource
                     .newBuilder()
                     .setName("mem")
                     .setType(Value.Type.SCALAR)
                     .setRole(memRole)
-                    .setScalar(Value.Scalar.newBuilder().setValue(taskMem - containerMem)))
+                    .setScalar(Value.Scalar.newBuilder().setValue(taskMem)))
```

Comment:
Perhaps we should think about splitting out some of this memory to the Task below. I had trouble before doing this and causing out of memory errors, but it's worth a shot.